### ISSUE TITLE
Bundler: BUNDLE_DISABLE_MULTISOURCE changes the Gemfile.lock + breaks deployment mode

### ISSUE THREAD

**Issue by grosser** [rubygems/rubygems#3643]
without BUNDLE_DISABLE_MULTISOURCE all gems are under a single source
with BUNDLE_DISABLE_MULTISOURCE gems are split into different sources

... prefer to always use the same approach so users do not run into merge conflicts and deployment mode install does not fail

workaround is using this at the top of the Gemfile:
```ruby
# https://github.com/rubygems/rubygems/issues/3643
Bundler::FeatureFlag.send(:settings_flag, :disable_multisource) { true }
```

**Comment by deivid-rodriguez** [comment]
This should get fixed by #4381!

**Comment by deivid-rodriguez** [comment]
The approach is that we'll start _always_ locking GEM sources separately, since it's the more secure "mode", but we'll keep the previous behaviour and print a warning when there's an existing lockfile with "merged sources".
